core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala

```diff
@@ -0,0 +1,156 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.server
+
+import kafka.utils.TestUtils
+import org.apache.kafka.common.internals.TopicConstants
+import org.apache.kafka.common.protocol.{ApiKeys, Errors}
+import org.apache.kafka.common.requests.{MetadataRequest, MetadataResponse}
+import org.junit.Assert._
+import org.junit.{Before, Test}
+
+import scala.collection.JavaConverters._
+
+class MetadataRequestTest extends BaseRequestTest {
+
+  def generateConfigs() = {
+    val props = TestUtils.createBrokerConfigs(numBrokers, zkConnect, enableControlledShutdown = false)
+    props.foreach { p =>
+      p.setProperty(KafkaConfig.DeleteTopicEnableProp, "true")
+      p.setProperty(KafkaConfig.RackProp, s"rack/${p.getProperty(KafkaConfig.BrokerIdProp)}")
+    }
+    props.map(KafkaConfig.fromProps)
+  }
+
+  private def allMetadataRequest = new MetadataRequest(null, true)
+  private def noMetadataRequest = new MetadataRequest(List[String]().asJava)
+
+  @Before
+  override def setUp() {
+    super.setUp()
+  }
+
+  @Test
+  def testControllerId() {
+    val metadataResponse = sendMetadataRequest(allMetadataRequest, 1)
+    val controllerServer = servers.find(_.kafkaController.isActive()).get
+    val controllerId = controllerServer.apis.brokerId
+
+    assertEquals("Controller id should match the active controller",
+      controllerServer.apis.brokerId, controllerId)
```

Comment:
This is doing the same thing as the `controllerId` line. I think you meant to get the controller id from `metadataResponse`.